### DISCUSSION TITLE
Timestamp issues with using --language English to force transcription of non-English audio.

### DISCUSSION THREAD

**Discussion by despairTK** [openai/whisper#1862] [Q&A]
This problem is mainly for transcribing non-English audio. I have tested Japanese audio, Spanish audio, and Korean audio.
If the language specified when generating subtitles is the same as the language of the transcribed audio (such as Japanese audio, set --language Japanese), the subtitle results will be There will be some selection errors for polysyllabic words.
However, if you force the generated subtitles to be in English (such as Japanese audio, set --language English), you will get more accurate subtitle results than the original language of the audio.
And the results obtained by using --language English to force the transcription of non-English audio are very different from --task translate, and are more accurate than --task translate.
However, the timestamp may have incorrect positions or be of too short duration, and there may also be repeated hallucinations.

It would be even better if the timestamps could be normal when using --language English to force transcribe non-English audio. Otherwise, you can only manually correct the timestamp of each line of subtitles.

Hallucinations and timestamp issues that may occur:
![287414713-3128f507-c29a-4ee7-b584-75e5388ad6bc](https://github.com/openai/whisper/assets/111871110/a65f75a9-8632-492e-8a17-46d54a8062fa)

**Comment by Purfview** [comment]
He says that for the non-English audios setting a hacky `--language English`  gets more accurate results than using `--task=translate`.

@despairTK  Do you mean that you don't have inaccurate timestamps with `--task=translate`?

**Reply by despairTK** [reply]
--task=translate will also result in inaccurate timestamps.
